app/models/state_routing_target.rb

```diff
@@ -15,6 +15,8 @@
 #
 class StateRoutingTarget < ApplicationRecord
   belongs_to :target, polymorphic: true
+
+  validates :state_abbreviation, uniqueness: { scope: [:target] }
```

Comment:
This is good.

Nit: I would also like to see the corresponding SQL constraint.